numpy/ma/core.py

```diff
@@ -175,6 +176,31 @@ class MaskError(MAError):
                   'V': b'???',
                   'U': 'N/A'
                   }
+"""
+
+"""
+Addressing issue 25677
+Added different custom fill values for different ints and floats.
+"""
+default_filler = {'b': True,
+                  'c': 1.e20 + 0.0j,
+                  "float16": 65504,
+                  "float32": pow(2, 31) - 1,
+                  "float64": pow(2, 63) - 1,
+                  "int8": 127,
+                  "uint8": 255,
+                  'int16': 32767,
+                  "uint16": 65535,
+                  "int32": 2147483647,
+                  "uint32": 4294967295,
+                  "int64": 9223372036854775807,
+                  "uint64": pow(2, 64) - 1,
```

Comment:
I think all of the integer dtypes that _can_ represent `999999` should retain that value. Otherwise, we'd have to change all of those tests that are now failing. Similarly, the floating point dtypes should retain the old value.

Reply:
Ok, thanks for letting me know. I reverted the int types back as follows:

`default_filler = {'b': True,
                  'c': 1.e20 + 0.0j,
                  "float16": 65504,
                  "float32": 1.e20,
                  "float64": 1.e20,
                  "i": 999999,
                  'O': '?',
                  'S': b'N/A',
                  'V': b'???',
                  'U': 'N/A'
                  }`

However, float16 remains 65504 to avoid the inf value per the original issue (https://github.com/numpy/numpy/issues/25677).

Does this make sense?

Reply:
#25677 is still unresolved for the smaller integer types. Write out the unit tests that test for the behavior that you want. Demonstrate (to yourself) that they fail without your changes, then add your changes until the tests pass.